torat_server/client_ishell.go

```diff
@@ -277,6 +282,27 @@ func (ac *activeClient) Reconnect(c *ishell.Context) {
 	c.Stop()
 }
 
+// Hardware print clients Operating-system info
+func (ac *activeClient) OSInfo(c *ishell.Context) {
+	c.ProgressBar().Indeterminate(true)
+	c.ProgressBar().Start()
+	r := shared.OSInfo{}
+
+	if err := ac.RPC.Call("API.GetOSInfo", void, &r); err != nil {
+		c.ProgressBar().Final(yellow("["+ac.Data().Name+"] ") + red("[!] Could not collect information on client OSInfo:", err))
+		c.ProgressBar().Stop()
+		return
+	}
+
+	c.ProgressBar().Final(yellow("["+ac.Data().Name+"] ") + green("[+] OSInfo collection finished"))
+	c.ProgressBar().Stop()
+
+	c.Println(green("Runtime:     "), r.Runtime)
+	c.Println(green("Arch:        "), r.OSArch)
+	c.Println(green("OS:          "), r.OSName)
+	c.Println(green("OS Version:  "), r.OSVersion)
+}
+
```

Comment:
Hardware does sth very similar why do you want this?

Reply:
We keep running into errors with GHW. This is to make the API for scripting more stable since on windows clients we seem to be crashing. (already raised issues upstream)